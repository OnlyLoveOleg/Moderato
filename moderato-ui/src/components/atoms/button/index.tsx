export * from './button';
export * from './hamburger';
export * from './rectangle';
