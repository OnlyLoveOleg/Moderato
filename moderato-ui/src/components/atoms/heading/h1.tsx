import { NextComponentType, NextPageContext } from 'next';
import styled from 'styled-components';

export type Props = {
  className?: string; // 親でstyleを当てれるようにするため
  text: string;
  size: string;
};

const StyledH1 = styled.h1<Omit<Props, 'text'>>`
  font-size: ${(props) => props.size};
  letter-spacing: 0.05em;
`;

export const H1: NextComponentType<NextPageContext, null, Props> = ({
  className = '',
  text,
  size,
}) => {
  return (
    <StyledH1 className={className} size={size}>
      {text}
    </StyledH1>
  );
};
