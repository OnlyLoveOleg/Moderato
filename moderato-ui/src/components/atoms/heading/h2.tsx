import { NextComponentType, NextPageContext } from 'next';
import styled from 'styled-components';

type Props = {
  text: string;
  size: string;
};

const StyledH2 = styled.h2<Omit<Props, 'text'>>`
  font-size: ${(props) => props.size};
  letter-spacing: 0.05em;
`;

export const H2: NextComponentType<NextPageContext, null, Props> = ({ text, size }) => {
  return <StyledH2 size={size}>{text}</StyledH2>;
};
