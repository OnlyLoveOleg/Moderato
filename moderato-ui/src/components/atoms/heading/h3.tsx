import { NextComponentType, NextPageContext } from 'next';
import styled from 'styled-components';

type Props = {
  text: string;
  size: string;
};

const StyledH3 = styled.h3<Omit<Props, 'text'>>`
  font-size: ${(props) => props.size};
  letter-spacing: 0.05em;
`;

export const H3: NextComponentType<NextPageContext, null, Props> = ({ text, size }) => {
  return <StyledH3 size={size}>{text}</StyledH3>;
};
