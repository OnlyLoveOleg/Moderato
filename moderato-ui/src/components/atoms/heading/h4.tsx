import { NextComponentType, NextPageContext } from 'next';
import styled from 'styled-components';

type Props = {
  className?: string;
  text: string;
  size: string;
};

const StyledH4 = styled.h4<Omit<Props, 'text'>>`
  font-size: ${(props) => props.size};
  letter-spacing: 0.05em;
`;

export const H4: NextComponentType<NextPageContext, null, Props> = ({
  className = '',
  text,
  size,
}) => {
  return (
    <StyledH4 className={className} size={size}>
      {text}
    </StyledH4>
  );
};
