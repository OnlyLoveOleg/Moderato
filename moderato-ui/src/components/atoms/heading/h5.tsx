import { NextComponentType, NextPageContext } from 'next';
import styled from 'styled-components';

type Props = {
  className?: string;
  text: string;
  size: string;
};

const StyledH5 = styled.h5<Omit<Props, 'text'>>`
  font-size: ${(props) => props.size};
  letter-spacing: 0.05em;
`;

export const H5: NextComponentType<NextPageContext, null, Props> = ({
  className = '',
  text,
  size,
}) => {
  return (
    <StyledH5 className={className} size={size}>
      {text}
    </StyledH5>
  );
};
