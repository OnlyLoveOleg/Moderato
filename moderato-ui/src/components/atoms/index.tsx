export * from './Title';
export * from './button';
