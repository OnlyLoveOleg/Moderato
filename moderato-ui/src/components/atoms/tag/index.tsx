// recommend で使うtag
export * from './category';
export * from './icon-tag';
