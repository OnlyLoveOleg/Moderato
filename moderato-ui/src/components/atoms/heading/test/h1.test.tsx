import { render, screen } from '@testing-library/react';
import { H1 } from '@/components/atoms/heading';
import { screenShot } from '@/tests/util';

describe('Heading H1 test', () => {
  test('Check H1 view', () => {
    render(<H1 text='naohito-T' size='48px' />);
    expect(screen.getByText('naohito-T'));
  });

  test('Check snapshot', () => {
    const tree = screenShot(<H1 text='naohito-T' size='48px' />);
    expect(tree).toMatchSnapshot();
  });
});
