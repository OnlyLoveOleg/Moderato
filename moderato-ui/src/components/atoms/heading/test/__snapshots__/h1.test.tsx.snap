// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading H1 test Check snapshot 1`] = `
<h1
  className="sc-a81cf63e-0 joWpYe"
  size="48px"
>
  naohito-T
</h1>
`;
